src/find_missing_words/gui/search.py

```diff
@@ -128,17 +126,20 @@ def search(self):
             note_creation_window.show()
         else:
             self.note_creation_window.word_select.set_word_model(word_model)
-            self.note_creation_window.word_select.reset()
+            self.note_creation_window.word_select.build()
 
     def build_word_model(self, text):
         word_model = {}
-        tokens = re.findall(utils.token_regex, text)
+        tokens = utils.split_words(text)
         for token in tokens:
+            if not utils.is_word(token):
+                continue
             query = self.get_final_search(token)
             found_note_ids = mw.col.findNotes(query)
             config = mw.addonManager.getConfig(__name__)
             ignored_words = config[ConfigProperties.IGNORED_WORDS.value]
-            known = len(found_note_ids) > 0 or token.lower() in [ignored_word.lower() for ignored_word in ignored_words]
+            is_word = utils.is_word(token)
```

Comment:
Repeated effort, move variable to top of for loop so it is not done twice. Actually... :thinking: if this is false it'll never get into here anyways :100: :thinking:

Reply:
Good catch, wrote this line before the other one and forgot about it.

Reply:
I'll also add some comments!